packages/shared/lib/validator.ts

```diff
@@ -582,6 +583,103 @@ class TypeValidator extends Validator {
     }
 }
 
+/**
+ * Validation for get node info
+ */
+ class NodeInfoValidator extends Validator {
+    /**
+     * Checks if response is valid
+     *
+     * @method isValid
+     *
+     * @param {MessageResponse} response
+     *
+     * @returns {ValidationResponse}
+     */
+    isValid(response: MessageResponse): ValidationResponse {
+        const payload = response.payload as NodeInfo
+
+        if ('object' !== typeof payload.nodeinfo) {
```

Comment:
I'm not sure if this would ever happen in this case, but `typeof` will actually return `object` if a variable is null. Would that then result in `Cannot read property 'version' of null` later in this function?

Reply:
I have improved the check, as you said very unlikely to happen, but in the case we dont get a properly formatted response from a broken node or load balancer etc it should be handled.